include/maliput_sparse/builder/builder.h

```diff
@@ -0,0 +1,360 @@
+  SegmentBuilder& EndLane();
+
+  /// @brief Sets a maliput_sparse::geometry::LaneGeometry into this builder to fill in the Lane.
+  /// @details This method is only intended to be called by LaneGeometryBuilder instances.
+  /// @see maliput::common::Passkey class description for further details.
+  /// @param lane_geometry A maliput_sparse::geometry::LaneGeometry to be stored into the Lane. It must not be nullptr.
+  /// @throws maliput::common::assertion_error When @p lane_geometry is nullptr.
+  void SetLaneGeometry(maliput::common::Passkey<LaneGeometryBuilder>,
+                       std::unique_ptr<maliput_sparse::geometry::LaneGeometry> lane_geometry);
+
+ private:
+  maliput::api::LaneId id_{"unset_id"};
+  maliput::api::HBounds hbounds_{};
+  std::unique_ptr<maliput_sparse::geometry::LaneGeometry> lane_geometry_{};
+};
+
+/// @brief Builder class for maliput::api::Segments.
+class SegmentBuilder final : public details::NestedBuilder<JunctionBuilder> {
+ public:
+  /// @brief Construct a new Segment Builder object.
+  /// @param parent The parent JunctionBuilder. It must not be nullptr.
+  explicit SegmentBuilder(JunctionBuilder* parent) : details::NestedBuilder<JunctionBuilder>(parent) {}
+
+  /// @brief Sets the maliput::api::SegmentId of the maliput::api::Segment.
+  /// @param segment_id The maliput::api::SegmentId.
+  /// @return A reference to this SegmentBuilder.
+  SegmentBuilder& Id(const maliput::api::SegmentId& segment_id);
+
+  /// @brief Starts the Lane builder for this Segment.
+  /// @details Lanes should be added from right to left as the order they are set will determine the
+  /// index they will receive in consecutive natural numbers (0 to +inf). That order must follow
+  /// the indexing order required by the maliput::api::Segment::index() contract. This builder makes
+  /// no assertion of that.
+  /// @return A LaneBuilder.
+  LaneBuilder StartLane();
+
+  /// @brief Finalizes the construction process of this Segment by inserting the Segment into the
+  /// parent JunctionBuilder.
+  /// @throws maliput::common::assertion_error When there is no lane to be set into the Segment.
+  /// @return A reference to the JunctionBuilder.
+  JunctionBuilder& EndSegment();
+
+  /// @brief Sets a maliput::geometry_base::Lane into this builder to fill in the Segment.
+  /// @details This method is only intended to be called by LaneBuilder instances. Call this method
+  /// in order to determine a specific left to right ordering for this Segment.
+  /// @see maliput::common::Passkey class description for further details.
+  /// @param lane A lane to be stored into the Segment. It must not be nullptr.
+  /// @throws maliput::common::assertion_error When @p lane is nullptr.
+  void SetLane(maliput::common::Passkey<LaneBuilder>, std::unique_ptr<maliput::geometry_base::Lane> lane);
+
+ private:
+  maliput::api::SegmentId id_{"unset_id"};
+  std::vector<std::unique_ptr<maliput::geometry_base::Lane>> lanes_{};
```

Comment:
Use `maliput_sparse::Lane` instead

Reply:
I did not want to include maliput::Lane because we need to expose the type and the SegmentBuilder actually does not care about the maliput_sparse::Lane but maliput::geometry_base::Lane.